### ISSUE TITLE
Display only causal models on GO-CAM site

### ISSUE THREAD

**Issue by dustine32** [geneontology/web-gocam#20]
As suggested by @thomaspd, we should only show the most relevant (i.e., causal) GO-CAM models on the GO-CAM browser site.

I believe a fast way (perhaps we could call it a hack) to implement this is to just add the `?causlmf=2` parameter to [the API call](https://github.com/geneontology/pipeline/blob/b97a53e46a7f87391e61e924ac5666603bcf9d0a/Jenkinsfile#L465) used to generate the cached `gocam-models.json` file. The other three files shouldn't require any change as they're just lookup files for the main `gocam-models.json` file.

**Comment by kltm** [comment]
Ah yes, here's the other one. Let's talk when you're back on making sure we're on the same page with https://github.com/geneontology/minerva/issues/448 .

**Comment by ValWood** [comment]
Whre is the GO-CAM browser. site?

**Comment by pgaudet** [comment]
[GO-CAM browser site](https://geneontology.cloud/home)

**Comment by kltm** [comment]
@dustine32 has a possibly easy upfront fix with #302. Testing now.

**Comment by kltm** [comment]
@pgaudet We may wrap this before even starting up. With the fix that @dustine32 has, http://geneontology.org/go-cam is not displaying 217 models and is quite snappy. Is this a sufficient fix for now?

**Comment by ValWood** [comment]
Cool, I had totally forgotten about this.
Note that the  link to the models from the "Browse models" tab is broken.

http://noctua.geneontology.org/editor/graph/gomodel:5745387b00001516

**Comment by pgaudet** [comment]
Can you explain what this is doing ? 

217 models seem a bit modest to me - is that all the connected models we have? 

- I would expect this one to be in: gomodel:6246724f00001921
It does show in Alliance: https://www.alliancegenome.org/gene/HGNC:20749
but when I search for gene name in the GO-CAM browser (ZDHHC20)I get 0 results. 

- Also, I would expect gomodel:60ad85f700000058 NOT to be in; there some connected activities, but these are binding, and especially, there are many unconnected activities. Interestingly, this one is NOT in the mouse Alliance Atf2 page: https://www.alliancegenome.org/gene/MGI:109349

Are we using different rules than Alliance? 


Thanks, Pascale

**Comment by pgaudet** [comment]
@pmasson55 estimates that about 100 Swiss-Prot models might be missing

**Comment by ValWood** [comment]
I thought it was low. I tried to browse by species. "elegans" gave 2 models.

**Comment by dustine32** [comment]
@pgaudet In general, it's querying all models containing a chain of at least two consecutive causal relations ("causally upstream of or within" RO:0002418 or descendants) connecting MF nodes. The actual query is [here](https://github.com/geneontology/api-gorest-2021/blob/965cba7dccb524d88586cf996ffe1e8715c92c14/queries/sparql-models.js#L322-L387).
* For gomodel:6246724f00001921 ZDHHC20-ZDHHC9 Palmitoyltransferase activity of Sars-Cov-2 S protein (Human-Sars-Cov-2), correct, this should be included in the GO-CAM browser site but for some reason is not picked up. And oddly it is showing up in the "by gene" query used by the Alliance gene page.
* For gomodel:60ad85f700000058 Atf2 mouse, I can see why this model is picked up by the causal query:
![image](https://user-images.githubusercontent.com/2678599/194107665-7e2ad2ec-0f83-44d9-9f12-58f9c97a303f.png)
It's kind of hard to see what the edges here are but they are "causally upstream of", and this chain of at least two consecutive causal relations between MFs is all that the query requires. Though, again, I agree that it's strange this model is not displayed on the Alliance gene page for [Atf2 Mmus](https://www.alliancegenome.org/gene/MGI:109349#pathways).

The [query](https://github.com/geneontology/api-gorest-2021/blob/965cba7dccb524d88586cf996ffe1e8715c92c14/queries/sparql-gp.js#L67-L157) to retrieve models "by gene" for the Alliance site should be very similar to the "all causal models" query used for the GO-CAM browser site, but your examples here show that there is some difference that needs to be fixed. I'll make a ticket in the [api-gorest-2021](https://github.com/geneontology/api-gorest-2021) repo to debug this. @pgaudet Thanks for the examples!

**Comment by kltm** [comment]
@dustine32 I wanted to follow up on this a little and make this is current. Looking at our current script, we have:
```
wget http://localhost:8888/models?causalmf=2 -O gocam-models.json'
```
that is generating the json that gets put into the release. This then gets propagated out (automatically) to the API now defined by https://github.com/geneontology/api-gorest-2023, correct? While we have a more expansive version of this with tagging models in metadata, this first pass is done, correct?

**Comment by dustine32** [comment]
@kltm Oh right, I believe this ticket is done.

And actually, this GO-CAM site pulls these JSONs directly from S3 rather than using a GO-CAM API passthrough:
https://github.com/geneontology/web-gocam/blob/cf224799ac5a644d104faa427f3695d4e8f6148c/src/app/core/gorest.service.ts#L37

**Comment by kltm** [comment]
@dustine32 yes, right you are; pushed on this bonus release pipeline: https://github.com/geneontology/pipeline/blob/f068cce4a3c1b84869e2f4cf501ac5b6c57e2e52/Jenkinsfile#L509
